tests/phpunit/tests/webfonts-api/wpWebfontsSchemaValidator.php

```diff
@@ -0,0 +1,470 @@
+					'font-weight' => 'invalid',
+				),
+				'expected' => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-style'   => 'normal',
+					'font-weight'  => '400',
+					'font-display' => 'fallback',
+				),
+			),
+			'font-display: invalid value'      => array(
+				'webfont'  => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-display' => 'invalid',
+				),
+				'expected' => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-style'   => 'normal',
+					'font-weight'  => '400',
+					'font-display' => 'fallback',
+				),
+			),
+		);
+	}
+
+	/**
+	 * @covers       WP_Webfonts_Registry::set_valid_properties
+	 *
+	 * @dataProvider data_set_valid_properties_with_invalid_and_error
+	 *
+	 * @param array  $webfont          Webfont input.
+	 * @param array  $expected         Expected updated webfont.
+	 * @param string $expected_message Expected notice message.
+	 */
+	public function test_set_valid_properties_with_invalid_and_error( array $webfont, array $expected, $expected_message ) {
+		$this->expectNotice();
+		$this->expectNoticeMessage( $expected_message );
+
+		$this->assertSame( $expected, self::$validator->set_valid_properties( $webfont ) );
+	}
+
+	/**
+	 * Data Provider.
+	 *
+	 * return @array
+	 */
+	public function data_set_valid_properties_with_invalid_and_error() {
+		return array(
+			'font-style: empty value'   => array(
+				'webfont'          => array(
+					'provider'    => 'some-provider',
+					'font-family' => 'Some Font',
+					'font-style'  => '',
+				),
+				'expected'         => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-style'   => 'normal',
+					'font-weight'  => '400',
+					'font-display' => 'fallback',
+				),
+				'expected_message' => 'Webfont font style must be a non-empty string.',
+			),
+			'font-weight: not a string' => array(
+				'webfont'          => array(
+					'provider'    => 'some-provider',
+					'font-family' => 'Some Font',
+					'font-weight' => null,
+				),
+				'expected'         => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-style'   => 'normal',
+					'font-weight'  => '400',
+					'font-display' => 'fallback',
+				),
+				'expected_message' => 'Webfont font style must be a non-empty string.',
+			),
+			'font-weight: empty value'  => array(
+				'webfont'          => array(
+					'provider'    => 'some-provider',
+					'font-family' => 'Some Font',
+					'font-weight' => '',
+				),
+				'expected'         => array(
+					'provider'     => 'some-provider',
+					'font-family'  => 'Some Font',
+					'font-style'   => 'normal',
+					'font-weight'  => '400',
+					'font-display' => 'fallback',
+				),
+				'expected_message' => 'Webfont font weight must be a non-empty string.',
+			),
+			'font-weight: not a string' => array(
```

Comment:
Same array key as in line 423

Reply:
Great catch @Soean! The first instance of that key was incorrect as it should have been `font-style`. Fixed in https://github.com/WordPress/wordpress-develop/pull/1736/commits/7939995d89f495560d29945b576292168084539f